drivers/led_strip/ws2812_spi.c

```diff
@@ -143,6 +147,10 @@ static int ws2812_strip_update_rgb(const struct device *dev,
 		}
 	}
 
+	/* Add stop frame to ensure MOSI stays low after transfer */
+	*px_buf = 0x00;
+	px_buf++;
```

Comment:
Because the low-to-high edge occurs right after the last frame, which some (the leds I have) detect as the start of another bit since the edge is within the T1+T0 time. This is received as a 1 bit (since the high edge is longer than T1H), whereas the stop byte ensures the low-to-high edge starts long after T1+T0 (about twice T1+T0)

Reply:
Oh this is actually a different behavior from the i.MX RT1064 MCU on which the MOSI signal keeps the state of the last sent value (from the logic analyzer capture attached to https://github.com/zephyrproject-rtos/zephyr/issues/66372). In your case the signal returns to a high level just after the transmission and it is interpreted as the next bit.

So what MCU model are you using ? Have you looked if it is possible to change the default level of the MOSI signal ? Maybe via some pinmux configuration ?